Redux 0.2.0

marz619 opened the pull request:
### NOTE: This contains breaking changes

### Changes

* `start_logger` renamed to `start_datadog_logger`
    + The returned function will first stop the internal QueueListener - which will attempt to push any remaining events to Datadog, followed by closing the DatadogHandler, which should closed the underlying Socket.

* `FormatFactory` renamed to `DatadogFormatter`
    + updates the logging attributes to be in line with Datadogs attributes/aliases documentation per their suggested naming conventions: [See here](https://docs.datadoghq.com/logs/log_configuration/attributes_naming_convention/)
        + `env` -> `dd.env`
        + `host` -> `dd.host`
        + `service` -> `dd.service`
        + `source` -> `dd.source`
        + `level` -> `status` 

* `DogHandler` renamed to `DatadogHandler` - no other internal changes, and does not need to be explicitly instantiated by a client, as the `start_datadog_logger` does this for us.

* Updated `README.md` to reflect the above changes
* Updated `LICENSE` date to 2023
* General formatting and ease of readability updates along with adding a basic `.pre-commit-config.yaml` file, and running it against the repo - did not result in significant changes, but found a few bugs through flake8/pylint.

marz619 committed:
Redux 0.2.0

NOTE: This contains breaking changes

* FormatFactory renamed to DatadogFormatFactory
+ updates the logging attributes to be inline with Datadogs
  attributes/aliases documentation

* DogHandler renamed to DatadogHandler - no other internal changes

General formatting and ease of readability updates

marz619 committed:
update LICENSE, README, and setup.py

marz619 committed:
add basic pre-commit + fixes after first run

jonprindiville commented on datadoglog/formatters.py:
```diff
@@ -1,47 +1,62 @@
+            (exc_type, exc_value, exc_tb) = exc_info
+            data["error.kind"] = str(exc_type)
+            data["error.message"] = str(exc_value)
+            data["error.stack"] = "\n".join(traceback.format_tb(exc_tb))
```
I saw this linked from elsewhere and was just a bit curious 👃...

I think the prior version of this line was correct: to be doing `"".join(...)` rather than `"\n".join(...)`.

Perhaps double-check in the specific Python version you're expecting to use, but 3.12 docs say the following about the output of `format_list` (which is used by `format_tb`):

> [...] return a list of strings ready for printing. [...] **Each string ends in a newline;** the strings may contain internal newlines as well, for those items whose source text line is not None.

-- https://docs.python.org/3.12/library/traceback.html#traceback.format_list

---

This jumped out at me because I had also been logging tracebacks recently. In my case I was using `format_stack` cause I didn't have an exception, but same difference I think: `format_stack` also relies on `format_list`.

(Speculative: if we wind up hitting some DD length limits and the joined TB strings gets truncated, could also probably leave it as a list of strings.)

marz619 replied:
Thanks for catching that, checked the docs for all versions 3.7+ and it's the same doc for the `format_list` method - I've reverted the change here: https://github.com/gadventures/datadoglog/pull/1/commits/fb38d28359821f50a9ad929e23e759071e7a0d03

marz619 committed:
clean up formatters some more ...

plus revert the "\n".join change back to "".join for the formatted
traceback attribute

marz619 committed:
return a closer that will stop the listener and close the handler
